src/Xrefcheck/Verify.hs

```diff
@@ -81,230 +87,324 @@ toVerifyRes = VerifyResult . either one (\() -> [])
+      -- some circumstances, we should do our best to report it as fine.
+      [ (403 ==)  -- unauthorized access
+      , (405 ==)  -- method mismatch
+      ]
 
     interpretErrors = \case
-        JsonHttpException _ -> error "External link JSON parse exception"
-        VanillaHttpException err -> case err of
-            InvalidUrlException{} -> error "External link URL invalid exception"
-            HttpExceptionRequest _ exc -> case exc of
-                StatusCodeException resp _
-                    | isAllowedErrorCode (statusCode $ responseStatus resp) -> Right ()
-                    | otherwise -> Left $ ExternalResourceUnavailable (responseStatus resp)
-                other -> Left . ExternalResourceSomeError $ show other
+      JsonHttpException _ -> error "External link JSON parse exception"
+      VanillaHttpException err -> case err of
+        InvalidUrlException{} -> error "External link URL invalid exception"
+        HttpExceptionRequest _ exc -> case exc of
+          StatusCodeException resp _
+            | isAllowedErrorCode (statusCode $ responseStatus resp) -> Right ()
+            | otherwise -> Left $
+                ExternalHttpResourceUnavailable (responseStatus resp)
+          other -> Left . ExternalResourceSomeError $ show other
+
+    checkFtp :: URI -> Bool -> ExceptT VerifyError IO ()
+    checkFtp uri secure = do
+      -- get authority which stores host and port
+      authority <- case uriAuthority uri of
+        Right a -> pure a
+        Left _ -> throwError ExternalResourceInvalidUri
+      let host = T.unpack . unRText $ authHost authority
+      let port :: Int = fromIntegral . fromMaybe 21 $ authPort authority
+      let pieces = uriPath uri
+      -- in case there aro no pieces - smth wrong with url
+      unless (isJust pieces) $
+        throwError ExternalResourceInvalidUri
+      let (trailing, pathList) = fromJust pieces
+      -- in case there are trailing slash - smth wrong with url
+      when trailing $
+        throwError ExternalResourceInvalidUri
+      -- build path from pieces
+      let path = T.unpack . mconcat
+            . NonEmpty.toList
+            . NonEmpty.intersperse "/"
+            . NonEmpty.map unRText
+            $ pathList
+      makeFtpRequest host port path secure `catch` \e ->
+        throwError $ ExternalFtpException e
+
+    makeFtpRequest
+      :: String
+      -> Int
+      -> FilePath
+      -> Bool
+      -> ExceptT VerifyError IO ()
+    makeFtpRequest host port path secure = handler host port $
+      \handle response -> do
+        -- check connection status
+        when (frStatus response /= Success) $
+          throwError $ ExternalFtpResourceUnavailable response
+        -- anonymous login
+        loginResp <- login handle "anonymous" ""
+        -- check login status
+        when (frStatus loginResp /= Success) $
+          throwError $ ExternalFtpResourceUnavailable loginResp
```

Comment:
This does not mean the resource is unavailable, it means the server does not support anonymous login.